packages/analytics/analytics-chart/src/components/DebugModal.vue

```diff
@@ -0,0 +1,139 @@
+<template>
+  <KTooltip
+    v-if="secretCodeEntered"
+    class="debug-tooltip"
+  >
+    <SupportIcon
+      class="debug-icon"
+      :size="KUI_ICON_SIZE_40"
+      @click="showDebugModal = true"
+    />
+    <template #content>
+      {{ i18n.t('debug') }}
+    </template>
+  </KTooltip>
+
+  <KPrompt
+    :is-visible="showDebugModal"
+    @canceled="showDebugModal = false"
+    @proceed="showDebugModal = false"
+  >
+    <template #body-content>
+      <KTabs :tabs="tabs">
+        <template #analytics-chart-data>
+          <KCodeBlock
+            v-if="analyticsChartData"
+            id="analyics-data-codeblock"
+            :code="analyticsChartData"
+            is-searchable
+            language="json"
+            max-height="250px"
+          />
+        </template>
+        <template #chart-js-data>
+          <KCodeBlock
+            v-if="chartData"
+            id="chartjs-data-codeblock"
+            :code="chartData"
+            is-searchable
+            language="json"
+            max-height="250px"
+          />
+        </template>
+        <template #chart-options>
+          <KCodeBlock
+            v-if="chartOptions"
+            id="options-codeblock"
+            :code="chartOptions"
+            is-searchable
+            language="json"
+            max-height="250px"
+          />
+        </template>
+      </KTabs>
+    </template>
+  </KPrompt>
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted, onUnmounted } from 'vue'
+import { SupportIcon } from '@kong/icons'
+import composables from '../composables'
+import { KUI_ICON_SIZE_40 } from '@kong/design-tokens'
+
+defineProps({
+  analyticsChartData: {
+    type: String,
+    required: true,
+  },
+  chartData: {
+    type: String,
+    required: true,
+  },
+  chartOptions: {
+    type: String,
+    required: true,
+  },
+})
+
+const { i18n } = composables.useI18n()
+const showDebugModal = ref(false)
+
+const tabs = [
+  {
+    hash: '#analytics-chart-data',
+    title: 'Chart Data',
+  },
+  {
+    hash: '#chart-js-data',
+    title: 'ChartJS Data',
+  },
+  {
+    hash: '#chart-options',
+    title: 'Chart Options',
+  },
+]
+
+const secretCode = ['ArrowUp', 'ArrowUp', 'ArrowDown', 'ArrowDown', 'ArrowLeft', 'ArrowRight', 'ArrowLeft', 'ArrowRight', 'KeyB', 'KeyA']
```

Comment:
What's the reasoning for shipping a `DebugModal` along with the component? Is this something we want to expose to customers? 

---

Separately, instead of keybindings, could we just look for a certain localStorage key? 

e.g. `localStorage.getItem(analytics-chart-debug, 'true')`?

Reply:
Its just a fun way to get the chart data out of a chart for a developer or power user. I thought using the code would be a fun way to hide it from customers.

Reply:
I'm not sure it's ideal to ship this into something that can be exposed in Production. Is that ideal? 

IMO we shouldn't be tying more keybindings; if I was working on another component in the Konnect UI and this conflicted, I'd have no way to know where it was coming from since it's not documented anywhere.

Reply:
@mihai-peteu why did you mark this as resolved?

Reply:
Feels like easter egg to me... I know fun stuff is fun, but...  
- Is this functionality something that is requested by PM?  
- Is there an agreement to document it and make it a part of product offering? 
- How often and by whom (other that author) will it be used? 

- If this is part of the functionality exposed to customers and documented - it should be easier accessible than trough the 10 key sequence, if this is for devs only ... hmm, I am confused as why it is needed .

Reply:
It's just a useful debugging tool, something that a power user or a developer would appreciate

Reply:
> something that a power user or a developer would appreciate

How would they ever know about it...? I'm not sure this belongs in Konnect?

Reply:
Also, what happens when there is more than one chart on the page?

Reply:
want to repeat my question: is this work done for fun or is it something product / pm requested?

Reply:
This seems like a component that could live and only be used in the [published sandbox](https://kongstrong.slack.com/archives/C01RF6STU2F/p1700171044487639) - wasn't that the point of us publishing the sandbox to Amplify?

Reply:
Just as an example. And this is something I've done many times to debug weird issues and edge cases that might come upt. 

But I imagine a use case where there's a weird issue with a particular chart in Konnect, you can use this to easily extract the chart data, then take that to the sandbox, past it in, in order to render that particular chart you saw in Konnect and start debugging. 

I can remove the debugger stuff for now until we run it by Christian first?

Reply:
> But I imagine a use case where there's a weird issue with a particular chart in Konnect, you can use this to easily extract the chart data, then take that to the sandbox, past it in, in order to render that particular chart you saw in Konnect and start debugging.

If there's an issue with a chart for a customer, wouldn't this data be sourced from Datadog, or by impersonating the user and inspecting what is going on with the request/response and how the UI is behaving?

IMO debugging tools have no place in the Production build/deployment.